opentracing-spring-web-handler-interceptor/src/test/java/io/opentracing/contrib/spring/web/servlet/TracingHandlerInterceptorTest.java

```diff
@@ -0,0 +1,256 @@
+package io.opentracing.contrib.spring.web.servlet;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import org.hamcrest.core.IsEqual;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.embedded.LocalServerPort;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.jayway.awaitility.Awaitility;
+
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.TextMapPropagator;
+import io.opentracing.tag.Tags;
+
+/**
+ * @author Pavol Loffay
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = {AppConfiguration.class,
+    WebSecurityConfig.class})
+public class TracingHandlerInterceptorTest {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    private int port;
+
+    @Before
+    public void beforeTest() throws InterruptedException {
+        AppConfiguration.mockTracer.reset();
+    }
+
+    @Test
+    public void contextPropagation() throws Exception {
+        {
+            HttpHeaders headers = new HttpHeaders();
+            headers.set(TextMapPropagator.HEADER_SPAN_ID, "1");
+            headers.set(TextMapPropagator.HEADER_TRACE_ID, "345");
+
+            HttpEntity<String> entity = new HttpEntity<>(headers);
+
+            restTemplate.exchange("/sync", HttpMethod.GET, entity, String.class);
+            Awaitility.await().until(reportedSpansSize(), IsEqual.equalTo(1));
```

Comment:
Interesting, I did something similar for APM, as I didn't find any libs that would do this. Remind me to replace our custom made "Wait#until()" by this :)